### PR TITLE
[TEST] Make testNotReadyOnBadFileSettings more reliable

### LINKED ISSUE #93036: [CI] ReadinessClusterIT testNotReadyOnBadFileSettings failing
**Build scan:**
https://gradle-enterprise.elastic.co/s/ff6b7vchgnbps/tests/:server:internalClusterTest/org.elasticsearch.readiness.ReadinessClusterIT/testNotReadyOnBadFileSettings

**Reproduction line:**
```
./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.readiness.ReadinessClusterIT.testNotReadyOnBadFileSettings" -Dtests.seed=CFBF9AB86C96D74 -Dtests.locale=lt -Dtests.timezone=Australia/North -Druntime.java=17
```

**Applicable branches:**
main, 8.6

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.readiness.ReadinessClusterIT&tests.test=testNotReadyOnBadFileSettings

**Failure excerpt:**
```
java.lang.AssertionError: (No message provided)

  at __randomizedtesting.SeedInfo.seed([CFBF9AB86C96D74:5569B5C2DBD78C34]:0)
  at org.junit.Assert.fail(Assert.java:86)
  at org.junit.Assert.assertTrue(Assert.java:41)
  at org.junit.Assert.assertTrue(Assert.java:52)
  at org.elasticsearch.readiness.ReadinessClusterIT.testNotReadyOnBadFileSettings(ReadinessClusterIT.java:267)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

### PR DESCRIPTION
**Pull request by grcevski**
The ReadinessClusterIT.testNotReadyOnBadFileSettings test expects to receive a cluster state update event, showing that we have recorded an error during processing of the file settings.

However, since we process these file settings on node start, the node can launch the file settings watcher thread and process everything before the test ever sets up the cluster state listener. Therefore, the test may never receive a cluster state update to unlock the await.

The fix introduced a harmless write of some cluster settings, to ensure that at least one cluster state event will be sent after we setup the cluster state listener.

Closes https://github.com/elastic/elasticsearch/issues/93036

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/readiness/ReadinessClusterIT.java [hunk 1]
```diff
@@ -319,4 +325,12 @@ public void testReadyAfterCorrectFileSettings() throws Exception {
         ReadinessService s = internalCluster().getInstance(ReadinessService.class, internalCluster().getMasterName());
         tcpReadinessProbeTrue(s);
     }
+
+    private void causeClusterStateUpdate() throws ExecutionException, InterruptedException {
+        ClusterUpdateSettingsRequest req = new ClusterUpdateSettingsRequest().persistentSettings(
+            Settings.builder().put("search.allow_expensive_queries", "false")
+        );
+        // This should succeed, it's never a reserved state in this test
+        client().admin().cluster().updateSettings(req).get();
```
Comment (DaveCTurner) [inline server/src/internalClusterTest/java/org/elasticsearch/readiness/ReadinessClusterIT.java:334]: I'd rather we just explicitly did a cluster state update that doesn't change anything, if that's the intention here:

```suggestion
        internalCluster().getCurrentMasterNodeInstance(ClusterService.class)
            .submitUnbatchedStateUpdateTask("poke", new ClusterStateUpdateTask() {
                @Override
                public ClusterState execute(ClusterState currentState) {
                    return ClusterState.builder(currentState).build();
                }

                @Override
                public void onFailure(Exception e) {
                    assert false : e;
                }
            });
```

Comment (DaveCTurner) [reply server/src/internalClusterTest/java/org/elasticsearch/readiness/ReadinessClusterIT.java:334]: NB edit^ (first suggestion was faulty)

Comment (grcevski) [reply server/src/internalClusterTest/java/org/elasticsearch/readiness/ReadinessClusterIT.java]: OK, sounds good. I tested this new approach with manually inserting a delay to cause the original issue, it works well now.

File: server/src/main/java/org/elasticsearch/readiness/ReadinessService.java [hunk 2]
```diff
@@ -194,6 +194,12 @@ protected void doStop() {
     // package private for testing
     synchronized void stopListener() {
         assert enabled(environment);
+
+        // don't spam the logs with repeated false, if we were never active
+        if (ready() == false) {
+            return;
+        }
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/readiness/ReadinessService.java:201]: I'm uncomfortable having this in the same commit as a test fix, at least not without much deeper analysis. If it's just about logging, I think it'd be best to revert it from this PR and do this separately so it doesn't get backported.

Comment (grcevski) [reply server/src/main/java/org/elasticsearch/readiness/ReadinessService.java]: Sounds good, I've reverted it and I'll follow-up with a new PR. Thanks!

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Pull request merged**